pkgs/development/python-modules/msprime/default.nix

```diff
@@ -0,0 +1,77 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, setuptools-scm
+, pythonOlder
+, gsl
+, numpy
+, newick
+, tskit
+, demes
+, pytest
+, pytest-xdist
+, scipy
+}:
+
+buildPythonPackage rec {
+  pname = "msprime";
+  version = "1.2.0";
+  format = "pyproject";
+  disabled = pythonOlder "3.8";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-YAJa2f0w2CenKubnYLbP8HodDhabLB2hAkyw/CPkp6o=";
+  };
+
+  nativeBuildInputs = [
+    setuptools-scm
+    gsl
+  ];
+
+  buildInputs = [
+    gsl
+  ];
+
+  propagatedBuildInputs = [
+    numpy
+    newick
+    tskit
+    demes
+  ];
+
+  nativeCheckInputs = [
+    pytest
+    pytest-xdist
+    scipy
+  ];
+
+  checkPhase = ''
+    runHook preCheck
+
+    # avoid adding the current directory to sys.path
+    # https://docs.pytest.org/en/7.1.x/explanation/pythonpath.html#invoking-pytest-versus-python-m-pytest
+    pytest -v --import-mode append \
```

Comment:
Please use the pytestCheckHook

Reply:
this is related to @natsukium's change https://github.com/NixOS/nixpkgs/pull/240183#discussion_r1247847237
From what I understood this is a workaround so that tests can be run in this specific case.

Reply:
So I've just tried using the flag `--import-mode append` inside `pytestFlagsArray` and that doesn't work with `pytestCheckHook`

Reply:
It seems that this is a fairly rare case where pytestCheckHook does not work due to Python's complex import system.

Reply:
pytestCheckHook is just some shell scripting around pytest. It should really work with this, this is more about making the ignores below easier to manage.